staging/src/k8s.io/apiserver/pkg/server/signal.go

```diff
@@ -47,8 +49,15 @@ func SetupSignalContext() context.Context {
 	go func() {
 		<-shutdownHandler
 		cancel()
-		<-shutdownHandler
-		os.Exit(1) // second signal. Exit directly.
+		if exitOnSecondSignal {
```

Comment:
is this a temporary fix?

Reply:
Temporary in the sense of a downstream carry.

See description for background.